src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,46 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's contact component in the address book.
+ * Guarantees: immutable; is valid as declared in its more specific child classes
+ */
+public class Contact {
+
+    public final String value;
+    private boolean isPrivate;
+	
+	/**
+     * Validates given contact component.
+     *
+     * @throws IllegalValueException if given contact component string is invalid.
+     */
+    public Contact(String contactComponent, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        contactComponent = contactComponent.trim();
+        this.value = contactComponent;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Overriding this requires something else rather than instanceOf(), as mentioned during tutorial. If you are comparing objects from subclasses, they are all instances of the parent object.